services/core-web/cypress/e2e/majorprojects.cy.ts

```diff
@@ -0,0 +1,79 @@
+describe("Major Projects", () => {
+    beforeEach(() => {
+        cy.login();
+
+        cy.get('[data-cy="home-link-button-major-projects"]', { timeout: 5000 }).click();
+
+        // .eq(1) selects the second row (0-based index).
+        cy.get("[data-cy=major-projects-table-open-button]", { timeout: 5000 })
+            .eq(1)
+            .find("button")
+            .click();
+
+
+        cy.get('a[data-cy="project-description-view-link"]', { timeout: 5000 }).click();
+
+        // Wait for the edit button to be visible and click it
+        cy.get('#project-summary-submit').then(($button) => {
+            $button[0].click();
+        });
+    });
+
+    it("should upload a document successfully", () => {
+        // Access the file input element and attach a file from the fixtures directory.
+        cy.get('input[type="file"]').scrollIntoView().attachFile('dummy.pdf');
+
+        // This long time is required because it takes quite a while for the file (11kb) to upload.
+        cy.wait(25000);
```

Comment:
I think in the documentation it recommended putting an alias on the request and then you can wait for it to complete

Reply:
Sure, that is ideal. I tried it and noticed random failures with no resolution.

Reply:
Updated it to now intercept, alias and wait. I still set the wait time to 25 seconds because of the time it takes to complete 3 service calls downstream.

Reply:
Had to change the code again because it was passing on my machine but failing continuously in the pipeline.